### PR TITLE
Improve SLA generation functions + resolve package issues

### PR DESCRIPTION
**Pull request by nhsbsa-mamcp**

### CODE REVIEW

File: .Rbuildignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 ^addressMatchR\.Rproj$
 ^\.Rproj\.user$
 ^LICENSE\.md$
+^.github$
```
Comment (nhsbsa-mamcp) [inline .Rbuildignore:4]: Should not be included in built package

File: DESCRIPTION [hunk 2]
```diff
@@ -21,26 +21,39 @@ Authors@R:
...
 			  email = "adam.ivison@nhs.net"
+			  ),
+	   person(given = "Mark",
+			  family = "McPherson",
+			  role = "ctb",
+			  email = "mark.mcpherson@nhsbsa.nhs.uk"
 			  )
 	  )
-Description: Helper package for matching care home address using AddressBase Plus 
+Description: Helper package for matching care home address using AddressBase Plus.
```
Comment (nhsbsa-mamcp) [inline DESCRIPTION:31]: Prevents R CMD Check warning about full sentence needed

File: DESCRIPTION [hunk 3]
```diff
@@ -21,26 +21,39 @@ Authors@R:
...
-    stringi
+    dplyr,
+    jsonlite,
+    magrittr,
+    nhsbsaR,
+    readr,
+    rlang,
+    stringi,
+    tibble,
+    utils
```
Comment (nhsbsa-mamcp) [inline DESCRIPTION:51]: Add and arrange all needed packages

File: DESCRIPTION [hunk 4]
```diff
@@ -21,26 +21,39 @@ Authors@R:
...
 Suggests: 
     rmarkdown,
     knitr,
     testthat (>= 3.0.0)
 Config/testthat/edition: 3
 VignetteBuilder: knitr
+Remotes:
+    https://github.com/nhsbsa-data-analytics/nhsbsaR
```
Comment (nhsbsa-mamcp) [inline DESCRIPTION:59]: Allows the nhsbsaR package to be pulled in when installing

File: R/calc_addressbase_plus_single_line_address.R [hunk 5]
```diff
@@ -106,129 +68,50 @@ calc_addressbase_plus_geo_single_line_address <- function(
...
-          test = !is.null(LOCALITY),
-          yes = paste0(LOCALITY, ", "),
-          no = ""
-        ),
-        ifelse(
-          test = !is.null(TOWN_NAME),
-          yes = paste0(TOWN_NAME, ", "),
-          no = ""
-        )
+      SAO_DASH = ifelse(
```
Comment (nhsbsa-mamcp) [inline R/calc_addressbase_plus_single_line_address.R:71]: To be able to add in dashes as appropriate, need to do this before using `unite.tbl`

File: R/calc_match_addresses.R [hunk 6]
```diff
@@ -14,6 +14,8 @@
 #' @param lookup_postcode_col Column containing the lookup postcode
 #' @param lookup_address_col Column containing the lookup single line address
 #'
+#' @importFrom rlang ':='
```
Comment (nhsbsa-mamcp) [inline R/calc_match_addresses.R:17]: Handles R CMD CHECK warning

File: R/calc_addressbase_plus_single_line_address.R [hunk 7]
```diff
@@ -5,6 +5,8 @@
 #' @param df AddressBase Plus DB table
 #' @param include_postcode Whether or not to include postcode. Default is FALSE.
 #'
+#' @importFrom rlang .data
```
Comment (nhsbsa-mamcp) [inline R/calc_addressbase_plus_single_line_address.R:8]: Handles R CMD CHECK warning

File: R/calc_match_addresses.R [hunk 8]
```diff
@@ -44,7 +46,7 @@ calc_match_addresses <- function(
       suffix = c("", "_LOOKUP"),
       copy = TRUE
     ) %>%
-    dplyr::select(-JOIN_ADDRESS)
+    dplyr::select(-.data$JOIN_ADDRESS)
```
Comment (nhsbsa-mamcp) [inline R/calc_match_addresses.R:49]: This and all other changes in this file are just adding the `.data` pronoun to handle R CMD CHECK warnings

File: R/upload_addressbase_plus_to_oracle.R [hunk 9]
```diff
@@ -132,7 +132,7 @@ upload_addressbase_plus_to_oracle <- function(con, path, pattern = NULL) {
   for (data_file in data_files) {
 
     # What date are we loading
-    file_name <- tail(x = strsplit(data_file[1], "/")[[1]], n = 1)
+    file_name <- utils::tail(x = strsplit(data_file[1], "/")[[1]], n = 1)
```
Comment (nhsbsa-mamcp) [inline R/upload_addressbase_plus_to_oracle.R:135]: Handle R CMD CHECK warning. No point importing for a single use.

File: R/zzz.R [hunk 10]
```diff
@@ -0,0 +1,45 @@
+if(getRversion() >= "2.15.1") utils::globalVariables(
```
Comment (nhsbsa-mamcp) [inline R/zzz.R:1]: Handles R CMD CHECK complaining about undeclared variables

File: R/zzz.R [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    "PAO_DASH",
+    "PAO_END_NUMBER",
+    "PAO_END_SUFFIX",
+    "STREET_DESCRIPTION",
+    "LOCALITY",
+    "TOWN_NAME",
+    "POSTCODE_LOCATOR",
+
+    # Created columns that do not work with .data pronoun
+    "SCORE"
```
Comment (nhsbsa-mamcp) [inline R/zzz.R:46]: `dplyr::slice_max ` does not seem to handle `.data`. Would expect it to, as the documentation says it uses data-masking.

Hence why https://github.com/nhsbsa-data-analytics/addressMatchR/pull/38/files#diff-46de35c1814e363c6cb14e19349e8a03ae1311fbba3f26349fa1a5e95edc3736R171 uses bare `SCORE`.

File: R/calc_match_addresses.R (hunk 8 above)
Comment (nhsbsa-mamcp) [reply R/calc_match_addresses.R:49]: ...plus a missed namespacing of `across`

**Comment by nhsbsa-adnsh** [comment]
housekeeping fine and changes work to DPA and GEO SLA creation work as intended

**Pull request merged**